sdk/eventhub/event-hubs/src/eventHubReceiver.ts

```diff
@@ -240,9 +255,20 @@ export class EventHubReceiver extends LinkEntity {
       }
     };
 
+    this._onAbort = () => {
+      const desc: string =
+        `[${this._context.connectionId}] The receive operation on the Receiver "${this.name}" with ` +
+        `address "${this.address}" has been cancelled by the user.`;
+      log.error(desc);
+      this._onError!(new Error(desc));
+    };
+
     this._onAmqpError = (context: EventContext) => {
       const receiver = this._receiver || context.receiver!;
       const receiverError = context.receiver && context.receiver.error;
+      if (this._aborter) {
+        this._aborter.removeEventListener("abort", this._onAbort);
+      }
```

Comment:
`_onAmqpClose` will be called right after `_onAmqpError` (that's why we have the detached() being called in _onAmqpClose)

So, we don't need to remove the listener here as you are already doing that in _onAmqpClose